### PR TITLE
Fixes #1938: Updated healthCheck for Posts service

### LINKED ISSUE #1938: Update posts healthcheck with Redis check
Every [Satellite microservice includes a `/healthcheck`](https://github.com/Seneca-CDOT/satellite#satelliteoptions), which can be used to determine if the service is running.  For example, the current posts healthcheck is:

https://dev.api.telescope.cdot.systems/v1/posts/healthcheck

To make this really useful, we also need to know if any dependent service is also healthy.  In the case of posts, this means Redis needs to be up and accepting commands.

Redis has a [ping](https://redis.io/commands/PING) command that we can call and make sure it returns OK.

### PR DESCRIPTION
**Pull request by HyperTHD**
## Issue This PR Addresses
This PR closes #1938 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

With Redis added to Satellite, I'm able to update the posts service's health check to check if redis is functioning by using the ping function from Redis. This PR also updates the routes for the posts service to use the createError function and call that function using the `next` function as part of the catch.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/posts/src/index.js
Comment (humphd) [inline src/api/posts/src/index.js:8]: You should return the promise directly

```js
healthCheck: () => redis.ping(),
```

**Review by humphd — CHANGES_REQUESTED** [review]
One small thing and this is ready.

Reading your code, I also realize that we need to add a way to tell the Redis connection to close just as we're shutting down.  Right now we don't expose a hook for this from Satellite, but we can add one.

File: src/api/posts/src/index.js
Comment (humphd) [inline src/api/posts/src/index.js:8]: You're not returning the promise here.  Should be: `healthCheck: () => redis.ping()`

**Review by humphd — CHANGES_REQUESTED** [review]
I think to get the unit tests to pass, you're going to have to move your `redis` instance out of `index.js` and into its own file?  Not sure, but `redis` being undefined in the tests is odd.

**Review by chrispinkney — DISMISSED** [review]
LGTM besides the nit pointed out by Dave.

**Comment by HyperTHD** [comment]
@humphd. I might need to wait for the shutdown option to be merged into Satellite. Even in it's own file, redis seems to be undefined which is super weird

**Comment by humphd** [comment]
I had an epiphany while on a walk today: did we add Redis since the last npm update for Satellite?  We probably need to ship an update that includes it.  I just triggered 1.12.0, which should hit npm within the next 30 mins.  Let's try that and see what happens here.

**Review by humphd — DISMISSED** [review]
This looks good.  The only minor nit I have is that you're only exporting a single thing from `redis.js`, so I would `module.exports = redis` instead, so you can `const redis = require('./redis');` and not bother with destructuring.  But it's minor.

I agree, we should do a follow-up to add the shutdown logic for redis too, once that lands.

File: src/api/posts/src/index.js
Comment (humphd) [inline src/api/posts/src/index.js:2]:
```js
const = redis = require('./redis');
```

File: src/api/posts/src/storage.js
Comment (humphd) [inline src/api/posts/src/storage.js:2]: Same thing here, no `{...}` anymore.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/posts/src/index.js
Comment (HyperTHD) [reply src/api/posts/src/index.js:2]: You mean 
```js
const redis = require('./redis'); 
```
right?

**Review by humphd — DISMISSED** [review]
Perfect.

**Review by chrispinkney — DISMISSED** [review]
LGTM

**Comment by HyperTHD** [comment]
Rebased, @humphd @chrispinkney, let's get this merged in

**Review by humphd — APPROVED** [review]

**Review by humphd — DISMISSED** [review]

**Pull request merged**
